server/service/orbit_client.go

```diff
@@ -537,38 +548,54 @@ func (oc *OrbitClient) getNodeKeyOrEnroll() (string, error) {
 	default:
 		return "", fmt.Errorf("read orbit node key file: %w", err)
 	}
-	var (
-		orbitNodeKey_        string
-		endpointDoesNotExist bool
-	)
+	var orbitNodeKey_ string
 	if err := retry.Do(
 		func() error {
-			var err error
 			orbitNodeKey_, err = oc.enrollAndWriteNodeKeyFile()
-			switch {
-			case err == nil:
-				return nil
-			case errors.Is(err, notFoundErr{}):
-				// Do not retry if the endpoint does not exist.
-				endpointDoesNotExist = true
-				return nil
-			default:
-				logging.LogErrIfEnvNotSet(constant.SilenceEnrollLogErrorEnvVar, err, "enroll failed, retrying")
-				return err
-			}
+			return err
 		},
 		// The below configuration means the following retry intervals (exponential backoff):
 		// 10s, 20s, 40s, 80s, 160s and then return the failure (max attempts = 6)
 		// thus executing no more than ~6 enroll request failures every ~5 minutes.
 		retry.WithInterval(orbitEnrollRetryInterval()),
 		retry.WithMaxAttempts(constant.OrbitEnrollMaxRetries),
 		retry.WithBackoffMultiplier(constant.OrbitEnrollBackoffMultiplier),
+		retry.WithErrorFilter(func(err error) (errorOutcome retry.ErrorOutcome) {
+			log.Info().Err(err).Msg("orbit enroll attempt failed")
+			switch {
+			case errors.Is(err, notFoundErr{}):
+				// Do not retry if the endpoint does not exist.
+				return retry.ErrorOutcomeDoNotRetry
+			case errors.Is(err, ErrEndUserAuthRequired):
+				// If we get an ErrEndUserAuthRequired error, then the user
+				// needs to authenticate with the identity provider.
+				//
+				// Open a browser window to the sign-on page and
+				// then keep retrying until they authenticate.
+				log.Debug().Msg("enroll unauthenticated, waiting for end-user to authenticate via SSO")
+				if !oc.initiatedIdpAuth {
+					oc.initiatedIdpAuth = true
+					log.Debug().Msg("opening SSO window")
+					openWindowErr := oc.openSSOWindow()
+					if openWindowErr != nil {
+						log.Error().Err(openWindowErr).Msg("opening SSO window")
+						return retry.ErrorOutcomeNormalRetry
+					}
+				}
+				// Sleep for 20 seconds, making the total retry interval 30 seconds
+				time.Sleep(20 * time.Second)
+				return retry.ErrorOutcomeResetAttempts
```

Comment:
Because we can't reliably close a browser window we've opened, for this first iteration we're just gonna open the window once per Orbit start. If the user closes it, they'll need to restart their computer to get it back. We may later implement a feature in Fleet Desktop to allow re-opening the SSO login window.

Reply:
Need to make sure to document this behavior.

While looking for the ticket to note this required documentation, I realized there was no subtask for docs on this story, so sounds like we need another subtask here.